frontend/js/src/common/brainzplayer/BrainzPlayer.tsx

```diff
@@ -193,8 +235,82 @@ export default class BrainzPlayer extends React.Component<
     ) {
       this.invalidateDataSource(this.soundcloudPlayer.current);
     }
+
+    // Fetch user's saved queue from localStorage
+    const savedQueue = localStorage.getItem("BrainzPlayer_queue");
+    if (savedQueue) {
+      try {
+        const {
+          userId,
+          queue,
+          queueRepeatMode,
+          currentListen,
+          currentDataSourceIndex,
+          currentTrackName,
+          currentTrackArtist,
+          currentTrackAlbum,
+          currentTrackURL,
+        } = JSON.parse(savedQueue) as BrainzPlayerQueueLocalStorage;
+        const { currentUser } = this.context;
+        if (userId === currentUser?.id) {
+          this.setState({
+            queue,
+            queueRepeatMode,
+            currentListen,
+            currentDataSourceIndex,
+            currentTrackName,
+            currentTrackArtist,
+            currentTrackAlbum,
+            currentTrackURL,
+          });
+        }
+      } catch (e) {
+        // Do nothing, we just fallback gracefully to the default queue.
+        const { listens } = this.props;
+        this.replaceQueue(listens);
+      }
+    } else {
+      const { listens } = this.props;
+      this.replaceQueue(listens);
```

Comment:
I don't think we want these fallback modes anymore, now that we have the ambient queue to replace it.

Currently this results in a duplicated list of listens in the queue and in the ambientQueue.

Reply:
Fixed